CONTRIBUTING.md

```diff
@@ -377,8 +377,22 @@ A problem must have a unique slug. This slug is used as
   this **must not** be merged until the exercism/x-common PR is merged.
 
 ## Improving Consistency By Extracting Shared Test Data
-
-TODO: elaborate.
+The goal is to have a file `exercises/<slug>/canonical-data.json` for each exercise in this repository. This JSON file should contain the generic (i.e. language independent) test data for that exercise. With this the quality of the tracks can be improved, e.g. fixes or new tests can be propagated to all the tracks. They can then be implemented manually or by running the test generator. (See [this issue](https://github.com/exercism/todo/issues/13) for more background info.)
+
+The following steps help you extracting the canonical test data.
+
+1. Select the exercise for which you want to create the shared test data from the [list of open todos](https://github.com/exercism/todo/issues?q=is%3Aissue+is%3Aopen+label%3Ashared-test-data)
+2. Now you need to find the existing tests. You can get a list with links to the all of those one of two ways
+  * Search for the exercise in this list http://x.exercism.io/v3/problems/data-todos (your browser might need a JSON viewer plugin/add-on to show this)
+  * Run this command in your terminal `curl http://x.exercism.io/v3/problems/data-todos | jq '.[] | map(select(.["slug"] == "accumulate"))'`. Replace `accumulate` with the name of the exercise you have chosen.
+3. Open the links in different browser tabs, navigate to the test files and read through the tests. Look for test cases that all/many of the languages have in common and identify outliers that only appear in one or two languages. It might be helpful to make some notes on your findings.
+4. Now it's time to write the canonical-data.json file. Look at examples like [the test data for hamming](https://github.com/exercism/x-common/blob/master/exercises/hamming/canonical-data.json) and check the [readme](https://github.com/exercism/x-common#test-data-format-canonical-datajson) to find out how to best structure the JSON file.
+From what you learned in step 3 decide for a sensible set of test cases. The following hints might help with that.
+  * Try to stick to test cases that already occur in many languages so it is feasible for the track maintainers to adapt to the canonical set.
+  * The tests should cover the described problem thoroughly enough but should not contain too many test cases that do not introduce any new aspect of the problem and already pass anyway.
+  * Make sure the described problem is tested. The test cases should not require people to follow one very specific way to solve the problem, e.g. the tests should avoid checking for helper functions that might not be present in other implementations.
+  * Performance tests should not be included in the canonical test data.
+5. Check the order of the tests. The simplest tests should be first.
```

Comment:
Is there anything specific about this that calls for a separate number? I ask because this seems to relate very closely to "should not contain too many test cases that do not introduce any new aspect of the problem and already pass anyway." - the main thing we might think about in test ordering is how easy it is to get to the first passing test. Then each test successively after that would ideally add a little functionality. All this so that the student can work iteratively and solve the problem a little at a time.